### PR TITLE
Implement sdk for Deta Drive

### PR DESCRIPTION
**Pull request by rabingaire**

### CODE REVIEW

File: src/drive/drive.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  }
+
+  /**
+   * get file from drive
+   *
+   * @param {string} name
+   * @returns {Promise<GetResponse>}
+   */
+  public async get(name: string): Promise<GetResponse> {
+    const trimedName = name.trim();
```
Comment (xeust) [inline src/drive/drive.ts:40]: `trimmedName`

File: src/drive/drive.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+    const { response, error } = await this.requests.delete(
+      DriveApi.DELETE_FILES,
+      payload
+    );
+    if (error) {
+      throw error;
+    }
+
+    return response?.deleted?.[0] || name;
```
Comment (xeust) [inline src/drive/drive.ts:85]: curious what the `?` does after a variable name in TypeScript? Never seen this syntax before and didn't find the docs.

File: src/drive/drive.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  }
+
+  /**
+   * list files from drive
+   *
+   * @param {ListOptions} [options]
+   * @returns {Promise<ListResponse>}
+   */
+  public async list(options?: ListOptions): Promise<ListResponse> {
+    const { prefix = '', limit = 1000, last = '' } = options || {};
```
Comment (xeust) [inline src/drive/drive.ts:130]: Do we want a limit of 1000 files?

File: src/drive/drive.ts (hunk 2 above)
Comment (rabingaire) [reply src/drive/drive.ts:85]: check this out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: src/drive/drive.ts [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+   * @param {string} contentType
+   * @returns {Promise<UploadResponse>}
+   */
+  private async upload(
+    name: string,
+    data: Buffer,
+    contentType: string
+  ): Promise<UploadResponse> {
+    const contentLength = data.byteLength;
+    const chunkSize = 1024 * 1024 * 100; // 100MB
```
Comment (aavshr) [inline src/drive/drive.ts:205]: we should decide on the chunk size, for the python sdk I switched back to `10`.

**Pull request merged**